lib/LaTeXML/Package/TeX.pool.ltxml

```diff
@@ -2294,12 +2294,12 @@ DefPrimitive('\read Number SkipKeyword:to SkipSpaces Token', sub {
     if (my $mouth = LookupValue('input_file:' . $port)) {
       $stomach->bgroup;
       AssignValue(PRESERVE_NEWLINES => 2);
-      AssignValue(INCLUDE_COMMENTS  => 0);
+      AssignValue(INCLUDE_COMMENTS  => 2);
       my @tokens = ();
       my ($t, $level) = (undef, 0);
       while ($t = $mouth->readToken) {
         my $cc = $t->getCatcode;
-        push(@tokens, $t);
+        push(@tokens, $t) unless $cc == CC_COMMENT;
```

Comment:
The spurious `\par` is due to `@tokens` being empty, even though we read "something".  A solution would be to add an `$ntokens` to the my, initialized to 0; increment it inside the `while`, and then a couple of lines below this code, change push's if test from `@tokens` to `$ntokens` (or code to the same effect).

Reply:
Once you pointed this out, it became completely obvious. Funny how that works. Fix is in - thanks!